NorthstarDLL/mods/modsavefiles.cpp

```diff
@@ -0,0 +1,232 @@
+	}
+
+	std::string fileName = g_pSquirrel<context>->getstring(sqvm, 1);
+	if (CheckFileName(fileName))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm,
+			fmt::format("File name invalid ({})! Make sure it has no '\\', '/' or non-ASCII charcters!", fileName, mod->Name).c_str());
+		return SQRESULT_ERROR;
+	}
+
+	std::string content = EncodeJSON<context>(sqvm);
+	if (ContainsNonASCIIChars(content))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm, fmt::format("File contents may not contain non-ASCII characters! Make sure your strings are valid!", mod->Name).c_str());
+		return SQRESULT_ERROR;
+	}
+
+	fs::create_directories(fs::path(GetNorthstarPrefix()) / "saveData" / fs::path(mod->m_ModDirectory).filename());
+	std::ofstream fileStr(fs::path(GetNorthstarPrefix()) / "saveData" / fs::path(mod->m_ModDirectory).filename() / (fileName));
+	if (fileStr.fail())
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm, fmt::format("There was an error opening/creating file {} (Is the file name valid?)", fileName).c_str());
+		return SQRESULT_ERROR;
+	}
+	fileStr.write(content.c_str(), content.length());
+	fileStr.close();
+
+	return SQRESULT_NULL;
+}
+
+ADD_SQFUNC("string", NSLoadFile, "string file", "", ScriptContext::CLIENT | ScriptContext::UI | ScriptContext::SERVER)
+{
+	Mod* mod = g_pSquirrel<context>->getcallingmod(sqvm);
+	if (mod == nullptr)
+	{
+		g_pSquirrel<context>->raiseerror(sqvm, "Has to be called from a mod function!");
+		return SQRESULT_ERROR;
+	}
+	std::string fileName = g_pSquirrel<context>->getstring(sqvm, 1);
+	if (CheckFileName(fileName))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm,
+			fmt::format("File name invalid ({})! Make sure it has no '\\', '/' or non-ASCII charcters!", fileName, mod->Name).c_str());
+		return SQRESULT_ERROR;
+	}
+
+	std::ifstream fileStr(fs::path(GetNorthstarPrefix()) / "saveData" / fs::path(mod->m_ModDirectory).filename() / (fileName));
+	if (fileStr.fail())
+	{
+		g_pSquirrel<context>->pushstring(sqvm, "");
+		return SQRESULT_NOTNULL;
+	}
+
+	std::stringstream jsonStringStream;
+	while (fileStr.peek() != EOF)
+		jsonStringStream << (char)fileStr.get();
+	g_pSquirrel<context>->pushstring(sqvm, jsonStringStream.str().c_str());
+
+	return SQRESULT_NOTNULL;
+}
+
+ADD_SQFUNC("table", NSLoadJSONFile, "string file", "", ScriptContext::CLIENT | ScriptContext::UI | ScriptContext::SERVER)
+{
+	Mod* mod = g_pSquirrel<context>->getcallingmod(sqvm);
+	if (mod == nullptr)
+	{
+		g_pSquirrel<context>->raiseerror(sqvm, "Has to be called from a mod function!");
+		return SQRESULT_ERROR;
+	}
+
+	std::string fileName = g_pSquirrel<context>->getstring(sqvm, 1);
+	if (CheckFileName(fileName))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm,
+			fmt::format("File name invalid ({})! Make sure it has no '\\', '/' or non-ASCII charcters!", fileName, mod->Name).c_str());
+		return SQRESULT_ERROR;
+	}
+
+	std::ifstream fileStr(fs::path(GetNorthstarPrefix()) / "saveData" / fs::path(mod->m_ModDirectory).filename() / (fileName));
```

Comment:
Given how often the path `fs::path(GetNorthstarPrefix()) / "saveData" / fs::path(mod->m_ModDirectory)` is used, it might be worth making it into a header variable or something so that we don't have to deal with re-evaluating it all the time?

Not a big deal though, this shouldn't get hit very often